frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/task-sidebar/TaskSidebarWorkspace.tsx

```diff
@@ -20,28 +25,43 @@ const TaskSidebarWorkspace: React.FC<TaskSidebarWorkspaceProps> = (props) => {
     resourceWorkspace: { name: workspaceName, optional = false },
   } = props;
   const { t } = useTranslation();
-  const { setFieldValue } = useFormikContext<PipelineBuilderFormikValues>();
+  const { getFieldMeta, setFieldValue } = useFormikContext<PipelineBuilderFormikValues>();
 
   const dropdownWorkspaces = availableWorkspaces.filter((workspace) => !!workspace.name?.trim());
 
+  const currentLinkedWorkspaceName = getFieldMeta<WorkspaceLink>(name).value?.workspace;
+  const currentLinkedWorkspaceSelectable =
+    currentLinkedWorkspaceName &&
+    dropdownWorkspaces.some((resource) => resource.name === currentLinkedWorkspaceName);
+
   const options: FormSelectFieldOption[] = [
     {
       label: t('pipelines-plugin~Select workspace...'),
       value: '',
-      isDisabled: true,
```

Comment:
And same comment here.

Reply:
Changed this to `isDisabled: !optional` aka `No {{resourceType}} resource`

Force pushed, PTAL